app/src/main/java/com/heig/atmanager/User.java

```diff
@@ -324,6 +329,73 @@ private  Date getDateXDaysAgo(int numberOfDaysAgo){
         return todos;
     }
 
+    public ArrayList<GoalTodo> getDailyGoalTodo(){
+        ArrayList<GoalTodo> result = new ArrayList<>();
+        Calendar current = Calendar.getInstance();
+        Date currentDate = new Date();
+        current.setTime(currentDate);
+
+        for(Goal g : goals){
+            for(GoalTodo gt : g.getGoalTodos()){
+                if(Arrays.equals(Utils.getDayWeekMonthYear(gt.getDoneDate()), Utils.getDayWeekMonthYear(currentDate))
+                        && g.getInterval() == Interval.DAY)
+                    result.add(gt);
+            }
+        }
+
+        return result;
+    }
+
+    public ArrayList<GoalTodo> getWeeklyGoalTodo(){
+        ArrayList<GoalTodo> result = new ArrayList<>();
+        Calendar current = Calendar.getInstance();
+        Date currentDate = new Date();
+        current.setTime(currentDate);
+
+        for(Goal g : goals){
+            for(GoalTodo gt : g.getGoalTodos()){
+                if(Arrays.equals(Utils.getWeekMonthYear(gt.getDoneDate()), Utils.getWeekMonthYear(currentDate))
+                        && g.getInterval() == Interval.WEEK)
+                    result.add(gt);
+            }
+        }
+
+        return result;
+    }
+
+    public ArrayList<GoalTodo> getMonthlyGoalTodo(){
+        ArrayList<GoalTodo> result = new ArrayList<>();
+        Calendar current = Calendar.getInstance();
+        Date currentDate = new Date();
+        current.setTime(currentDate);
+
+        for(Goal g : goals){
+            for(GoalTodo gt : g.getGoalTodos()){
+                if(Arrays.equals(Utils.getMonthYear(gt.getDoneDate()), Utils.getMonthYear(currentDate))
+                    && g.getInterval() == Interval.MONTH)
+                    result.add(gt);
+            }
+        }
+
+        return result;
+    }
+
+    public ArrayList<GoalTodo> getYearlyGoalTodo(){
+        ArrayList<GoalTodo> result = new ArrayList<>();
+        Calendar current = Calendar.getInstance();
+        Date currentDate = new Date();
+        current.setTime(currentDate);
+
+        for(Goal g : goals){
+            for(GoalTodo gt : g.getGoalTodos()){
+                if(Utils.getYear(currentDate) == Utils.getYear(gt.getDoneDate()) && g.getInterval() == Interval.YEAR)
+                    result.add(gt);
+            }
+        }
+
+        return result;
```

Comment:
I think you should refactor to :
´´´getGoalTodo(Interval interval)´´´ instead of copy pasting the code 4 times